pkg/metrics/metrics.go

```diff
@@ -0,0 +1,112 @@
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/rancher/rancher/pkg/auth/util"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/config"
+	authV1 "k8s.io/api/authorization/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	prometheusMetrics = false
+
+	gcInterval = time.Duration(60 * time.Second)
+
+	clusterOwner = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: "cluster_manager",
+			Name:      "cluster_owner",
+			Help:      "Set 1 when cluster is owned, by this metrics you can find out which rancher server run user controllers for specific cluster",
+		},
+		[]string{"cluster", "owner"},
+	)
+)
+
+type metricsHandler struct {
+	k8sClient kubernetes.Interface
+	next      http.Handler
+}
+
+func NewMetricsHandler(scaledContext *config.ScaledContext, handler http.Handler) http.Handler {
+	return &metricsHandler{
+		k8sClient: scaledContext.K8sClient,
+		next:      handler,
+	}
+}
+
+func (h metricsHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
```

Comment:
slept on it, maybe a pointer more efficient since this thing will be called a bunch, feeling this could fall within the when in doubt clause